suite/tests/runcmp.cmake

```diff
@@ -62,54 +62,52 @@ endif ()
 file(READ "${cmp}" expect)
 
 # Convert file contents into a CMake list, where each element in the list is one
-# line of the file, empty lines are ignored.
-string(REGEX REPLACE "^[\n]+" "" output "${output}")
-string(REGEX REPLACE "[\n]+$" "" output "${output}")
-string(REPLACE "\n\n" "\n" output "${output}")
-STRING(REGEX REPLACE ";" "\\\\;" output "${output}")
-STRING(REGEX REPLACE "\n" ";" output "${output}")
+# line of the file.
+string(REGEX MATCHALL "([^\n]+)\n" output "${output}")
+string(REGEX MATCHALL "([^\n]+)\n" expect "${expect}")
 
-string(REGEX REPLACE "^[\n]+" "" expect "${expect}")
-string(REGEX REPLACE "[\n]+$" "" expect "${expect}")
-string(REPLACE "\n\n" "\n" expect "${expect}")
-STRING(REGEX REPLACE ";" "\\\\;" expect "${expect}")
-STRING(REGEX REPLACE "\n" ";" expect "${expect}")
-
-if(NOT "${ignore_matching_lines}" STREQUAL "")
+if (NOT "${ignore_matching_lines}" STREQUAL "")
   # Filter out all the ignored lines.
   set(filtered_expect)
   foreach (item ${expect})
-      string(REGEX MATCH "${ignore_matching_lines}" match_result "${item}")
-      if (NOT match_result)
-          list(APPEND filtered_expect ${item})
-      endif ()
+    if (NOT "${item}" MATCHES "${ignore_matching_lines}")
+      list(APPEND filtered_expect ${item})
+    endif ()
   endforeach ()
 else ()
   set(filtered_expect ${expect})
 endif ()
 
-list (LENGTH filtered_expect filtered_expect_length)
-list (LENGTH output output_length)
+list(LENGTH filtered_expect filtered_expect_length)
+list(LENGTH output output_length)
 
 set(lists_identical TRUE)
 if (NOT ${filtered_expect_length} EQUAL ${output_length})
-    set(lists_identical FALSE)
+  set(lists_identical FALSE)
 endif ()
 
 if (lists_identical)
-  math(EXPR len "${filtered_expect_length} - 1")
-  foreach (index RANGE ${len})
-    list(GET filtered_expect ${index} item1)
-    list(GET output ${index} item2)
-    if (NOT ${item1} STREQUAL ${item2})
-        set(lists_identical FALSE)
-        message(STATUS "The first difference:")
-        message(STATUS "< ${item1}")
-        message(STATUS "> ${item2}")
-        break ()
+  if (filtered_expect_length GREATER "1000")
+    # CMake list(GET ...) operation is slow, do STREQUAL for large files, give up
+    # the diff output.
+    if(NOT filtered_expect STREQUAL output)
+      set(lists_identical FALSE)
     endif ()
-  endforeach ()
-endif ()
+  else ()
+    math(EXPR len "${filtered_expect_length} - 1")
+    foreach (index RANGE ${len})
```

Comment:
I think it'd be ok to bump the min cmake version to 3.17 (ubuntu20 has 3.27 at https://github.com/actions/runner-images/blob/main/images/linux/Ubuntu2004-Readme.md) and use ZIP_LISTS -- does that eliminate the slowness (your comment above)?

Reply:
I bumped CMake version to 3.17, but there are some CMake errors in Windows, which I don't understand, could you take a look?

Reply:
Never seen that -- did you try searching?  Searching shows solutions like https://github.com/RichieSams/FiberTaskingLib/issues/109.  But if it's too messy I would just bail and go back to the prior cmake with a XXX comment about switching to ZIP_LISTS when we've put in the effort elsewhere to get to 3.17.

Reply:
Yes I searched but didn't find out why.

> I would just bail and go back to the prior cmake with a XXX comment about switching to ZIP_LISTS when we've put in the effort elsewhere to get to 3.17.

Okay.